boards/x86/up_squared/doc/up_squared.rst

```diff
@@ -0,0 +1,238 @@
+
+The interrupt mode serial port provides general serial communication
+and external communication.
+
+Interrupt Controller
+--------------------
+
+This board uses the kernel's static Interrupt Descriptor Table (IDT) to program the
+Advanced Programmable Interrupt Controller (APIC) interrupt redirection table.
+
+
++-----+---------+--------------------------+
+| IRQ | Remarks | Used by Zephyr Kernel    |
++=====+=========+==========================+
+| 2   | HPET    | timer driver             |
++-----+---------+--------------------------+
+| 4   | UART    | serial port when used in |
+|     |         | interrupt mode           |
++-----+---------+--------------------------+
+
+HPET System Clock Support
+-------------------------
+
+Galileo uses HPET timing with legacy-free timer support. The board
+configuration uses HPET as a system clock timer.
+
+Connections and IOs
+===================
+
+Refer to the `UP Squared`_ website and `UP Squared Pinout`_ website
+for connection diagrams.
+
+Memory Mappings
+===============
+
+This board configuration uses default hardware memory map
+addresses and sizes.
+
+Programming and Debugging
+*************************
+
+Use the following procedures for booting an image on a Galileo board.
+
+* `Creating a GRUB2 Boot Loader Image from a Linux Host`_
+
+* `Preparing the Boot Device`_
+
+* `Booting the UP Squared Board`_
+
+Creating a GRUB2 Boot Loader Image from a Linux Host
+====================================================
+
+If you are having problems running an application using the preinstalled
+copy of GRUB, follow these steps to test on supported boards using a custom GRUB.
+
+#. Install the requirements to build GRUB on your host machine.
+
+   On Ubuntu, type:
+
+   .. code-block:: console
+
+      $ sudo apt-get install bison autoconf libopts25-dev flex automake
+
+   On Fedora, type:
+
+   .. code-block:: console
+
+     $ sudo dnf install gnu-efi bison m4 autoconf help2man flex \
+        automake texinfo
+
+#. Clone and build the GRUB repository using the script in Zephyr tree, type:
+
+   .. code-block:: console
+
+     $ cd $ZEPHYR_BASE
+     $ ./boards/x86/common/scripts/build_grub.sh x86_64
+
+#. Find the binary at
+   :file:`$ZEPHYR_BASE/boards/x86/common/scripts/grub/bin/grub_x86_64.efi`.
+
+Preparing the Boot Device
+=========================
+
+Prepare a USB flash drive to boot the Zephyr application image on
+a UP Squared board.
+
+#. Build a Zephyr application; for instance, to build the ``hello_world``
+   application on Galileo:
+
+   .. zephyr-app-commands::
+      :zephyr-app: samples/hello_world
+      :board: up_squared
+      :goals: build
+
+   .. note::
+
+      A stripped project image file named :file:`zephyr.strip` is automatically
+      created in the build directory after the application is built. This image
+      has removed debug information from the :file:`zephyr.elf` file.
+
+#. Refer to the UP Squared Serial Console Wiki page for instructions on how to
```

Comment:
Get the link working with:

    #. Refer to the `UP Squared Serial Console Wiki page
       <https://wiki.up-community.org/Serial_console>`_ for instructions on how to
       connect to a serial console.